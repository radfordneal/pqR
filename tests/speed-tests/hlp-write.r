source("time.r")
 
# Assess degree of parallelism using helper threads when writing to a file.

test.name <- "hlp-write"

g <- test.cmp (function ()
{ P <- M1 %*% M2
  write.table(P,file,quote=FALSE,row.names=FALSE,col.names=FALSE)
})

N <- 2
K <- 1500
M <- 100000

M1 <- matrix(c(1.1,0.2),N,K)
M2 <- matrix(c(4.1,2.2),K,M)

file <- tempfile(tmpdir=".")

g(); g()  # take time for memory allocation, etc. before test

options(helpers_disable=TRUE)
sys.time (g())

ans <- scan(file)
print(sum(ans))
print(sum(ans^2))

options(helpers_disable=FALSE)
sys.time (g())

stopifnot(identical(ans,scan(file)))

rm(ans)

unlink(file)
