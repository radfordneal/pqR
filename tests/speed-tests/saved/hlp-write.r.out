
---- hlp-write 

Time:
Read 200000 items
[1] 614250000
[1] 2.790703e+12
Time:
Read 200000 items
